src/Features/Core/Portable/Wrapping/InitializerExpression/InitializerExpressionCodeActionComputer.cs

```diff
@@ -0,0 +1,277 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Wrapping.SeparatedSyntaxList;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Wrapping.InitializerExpression
+{
+    internal abstract partial class AbstractInitializerExpressionWrapper<TListSyntax, TListItemSyntax>
+    {
+        private class InitializerExpressionCodeActionComputer : AbstractCodeActionComputer<AbstractInitializerExpressionWrapper<TListSyntax, TListItemSyntax>>
```

Comment:
Ditto to earlier question: this looks awfully close to SeparatedSyntaxListCodeActionComputer, can we reuse that somehow?

Reply:
agreed.

Reply:
Yea, I can look at how to reuse it. The main obstacle is after discussing with @CyrusNajmabadi we came to the conclusion to use a whole separate Wrapper class because the SeparatedSyntaxListWrapper is meant for argument/parameter lists. 

I could either make SeparatedSyntaxListCodeActionComputer a publicly available class to allow AbstractInitializerExpressionWrapper to use that code computer directly. Or I could take all the code in common between the two classes and put it into a third class they both inherent from it like below.

```C#
class InitializerExpressionCodeActionComputer : SeparatedListCodeActionComputer
{ ... }

class SeparatedSyntaxListCodeActionComputer : SeparatedListCodeActionComputer
{ ... }

public class SeparatedListCodeActionComputer
{ 
// All the common logic from InitializerExpressionCodeActionComputer & SeparatedSyntaxListCodeActionComputer  
}
```

I am not sure which option would be preferable, so I would welcome any feedback.